apps/site/assets/ts/schedule/ScheduleLoader.tsx

```diff
@@ -0,0 +1,241 @@
+import React, { ReactElement } from "react";
+import { connect } from "react-redux";
+import { useQueryParams, StringParam } from "use-query-params";
+import SchedulePage from "./components/SchedulePage";
+import ScheduleNote from "./components/ScheduleNote";
+import ScheduleDirection from "./components/ScheduleDirection";
+import { SchedulePageData, SelectedOrigin } from "./components/__schedule";
+import { MapData, StaticMapData } from "../leaflet/components/__mapdata";
+import ScheduleFinder from "./components/ScheduleFinder";
+import ScheduleFinderModal from "./components/schedule-finder/ScheduleFinderModal";
+import { DirectionId } from "../__v3api";
+import mapStateToProps, {
+  getCurrentState,
+  storeHandler
+} from "./ScheduleStore";
+
+interface Props {
+  schedulePageData: SchedulePageData;
+  component: string;
+  updateURL: Function;
```

Comment:
@alicia-mbta let's take advantage of the type system and write a more specific function type here, as described in https://www.typescriptlang.org/docs/handbook/functions.html#function-types